pybandits/quantitative_model.py

```
@@ -908,9 +908,11 @@ def _inner_update(self, segments: List[Segment], rewards: List[BinaryReward]):
         rewards : List[BinaryReward]
             Rewards for update.
         """
-        rewards = np.array(rewards)
-        for segment in set(segments):
-            rewards_of_segment = [r for r, s in zip(rewards, segments) if s == segment]
+        rewards_by_segment = defaultdict(list)
+        for segment, reward in zip(segments, rewards):
```

Ruff (0.15.2)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)